src/main/java/org/shrewsburyrobotics/pidlab/PIDResponseChart.java

```diff
@@ -39,35 +42,32 @@
 	private JTextField targetField = new JTextField("100", 4);
 	private JTextField durationField = new JTextField("10", 4);
 
-	private MotorModel motor = new MotorModel(Double.parseDouble(gainField.getText()),
-											  Double.parseDouble(timeField.getText()),
-											  Double.parseDouble(deadField.getText()));
+	private JRadioButton pButton = new JRadioButton("P");
+	private JRadioButton iButton = new JRadioButton("I");
+	private JRadioButton dButton = new JRadioButton("D");
+	private ButtonGroup pidSelector = new ButtonGroup();
 
-	private PIDController controller = new PIDController(Double.parseDouble(pField.getText()),
-														 Double.parseDouble(iField.getText()),
-														 Double.parseDouble(dField.getText()));
+	private ChartPanel chartPanel = new ChartPanel(createPIDChart());
 
-	private XYDataset dataset = createDataset(motor, controller, Double.parseDouble(targetField.getText()));
+	private XYDataset dataset = createDataset(query(pField), query(iField), query(dField), query(targetField));
 
 	public PIDResponseChart(String title) {
 		super(title);
 
-		// Create viewer panel in which to display the chart.
-		ChartPanel chartPanel = new ChartPanel(createPIDChart());
-
 		// Create controller panels where we read input values from.
-		JPanel motorPanel = createMotorPanel(); 
-		JPanel pidPanel = createPidPanel();
-		JPanel plotPanel = createPlotPanel(chartPanel);
+		JPanel UIPanel = new JPanel();
```

Comment:
I suggest rename UIPanel to controlsPanel. If you stick with UIPanel, it should be uiPanel.